sktime/transformations/series/peak.py

```diff
@@ -0,0 +1,441 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Extract peak/working hour features from datetimeindex."""
+__author__ = ["ali-parizad", "VyomkeshVyas"]
+__all__ = ["PeakTimeFeature"]
+
+
+import pandas as pd
+
+from sktime.transformations.base import BaseTransformer
+
+
+class PeakTimeFeature(BaseTransformer):
+    """PeakTime feature extraction for use in e.g. tree based models.
+
+    PeakTimeFeature uses a datetime index column and generates peak/working features
+    for e.g. peak hours, peak weak, peak month, working hours, etc. It works based on
+    the input intervals, e.g., peak_hour_start=[6], peak_hour_end=[9]
+
+    Parameters
+    ----------
+    ts_freq : str,
+        Restricts selection of items to those with a frequency lower than
+        the frequency of the time series given by ts_freq.
+        E.g., if daily data is provided and ts_freq = ("D"), it does not make
+        sense to derive PeakTimeFeature with higher frequency like hourly features. So,
+        the outpul will be is_peak_day, is_peak_week, is_peak_month, is_peak_quarter,
+        is_peak_year. Only supports the following frequencies:
+        * Y - year
+        * Q - quarter
+        * M - month
+        * W - week
+        * D - day
+        * H - hour
+
+    peak_[*]_start, peak_[*]_end : list,
```

Comment:
I understand this is boringly repetitive, but this is non-standard way to document arguments. Just noting this, if other revieweres have any comment to add.

Reply:
Thanks! updated in my next PR